### PR TITLE
support imported type aliases

### PR DESCRIPTION
**Pull request by reillywatson**
Go 1.23 changed how go/types handles type aliases, which breaks code generation if you use a package that has an alias type (example: https://pkg.go.dev/github.com/golang/protobuf@v1.5.3/ptypes/timestamp#Timestamp). These were getting generated as just the unqualified name, without the package. If you used a previous version of Go, it meant that if that aliased type was in an internal package we'd try to import the internal package instead, resulting in a compilation error.

Add a switch to handle that case, and bump the go.mod version to support using types.Alias.

### CODE REVIEW

File: pkg/moq/moq_test.go [hunk 1]
```diff
@@ -664,6 +664,22 @@ func TestImportedPackageWithSameName(t *testing.T) {
 	}
 }
 
+func TestImportedPackageWithTypeAlias(t *testing.T) {
```
Comment (sudo-suhas) [inline pkg/moq/moq_test.go:667]: Can you use the golden file test similar to https://github.com/matryer/moq/blob/v0.4.0/pkg/moq/moq_test.go#L409-L414?

Comment (reillywatson) [reply pkg/moq/moq_test.go]: Ah yeah I hadn't worked out how those golden files worked, that looks much better! Updated, thanks!

File: pkg/moq/testpackages/typealiastwo/internal/typealiasinternal/typealiasinternal.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package typealiasinternal
+
+// we shouldn't be able to import this type directly, you need to use the alias in the parent package
+type MyInternalType struct {
```
Comment (sudo-suhas) [inline pkg/moq/testpackages/typealiastwo/internal/typealiasinternal/typealiasinternal.go:4]: Does it make sense to test for a type with a type argument?

Comment (reillywatson) [reply pkg/moq/testpackages/typealiastwo/internal/typealiasinternal/typealiasinternal.go:4]: Yes, I think that's reasonable! Updated the test to include an alias with a type argument, as well as the original one without.

**Comment by sudo-suhas** [comment]
Thanks @reillywatson for the PR! Changes have been released in [moq@v0.5.0](https://github.com/matryer/moq/releases/tag/v0.5.0)

**Pull request merged**